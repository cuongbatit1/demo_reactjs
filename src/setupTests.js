// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import "@testing-library/react";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { REDUX_TOOL_KIT } from "./constants/AppConstants";
import configureStore from "./redux/configureStore";
// import configureStoreToolkit from "./reduxtoolkit/configureStoreToolkit";
import { MaterialUIControllerProvider } from "./context/MaterialUIControllerProvider";
import i18n from "./i18n";
import { BrowserRouter } from "react-router-dom";

function render(
  ui,
  {
    preloadedState,
    store = configureStore(preloadedState),
    // store = REDUX_TOOL_KIT ? configureStoreToolkit() : configureStore(),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    i18n.changeLanguage("en");
    return (
      <Provider store={store}>
        <MaterialUIControllerProvider>
          <BrowserRouter>{children}</BrowserRouter>
        </MaterialUIControllerProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
// override render method
export { render };
