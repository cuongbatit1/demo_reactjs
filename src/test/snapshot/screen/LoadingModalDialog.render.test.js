import { screen } from "@testing-library/react";
import LoadingModalDialog from "../../../ScreenPage/Dialog/LoadingModal";
import { render } from "../../../setupTests";

test("test render LoadingModalDialog snapshot", () => {
  const state = {
    loadingReducer: {
      isLoading: true,
    },
  };
  const { asFragment } = render(<LoadingModalDialog />, { preloadedState: state });
  expect(asFragment()).toMatchSnapshot();
});
