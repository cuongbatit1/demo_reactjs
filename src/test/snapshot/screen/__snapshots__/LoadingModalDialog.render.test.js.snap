// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render LoadingModalDialog snapshot 1`] = `<DocumentFragment />`;
